src/Controls/tests/DeviceTests/HandlerTestBase.cs

```diff
@@ -49,5 +51,15 @@ protected THandler CreateHandler<THandler>(IView view)
 
 		protected async Task<THandler> CreateHandlerAsync<THandler>(IView view) where THandler : IViewHandler =>
 			await InvokeOnMainThreadAsync(() => CreateHandler<THandler>(view));
+
+		protected Task<TValue> GetValueAsync<TValue, THandler>(IView view, Func<THandler, TValue> func)
+			 where THandler : IViewHandler
+		{
+			return InvokeOnMainThreadAsync(() =>
+			{
+				var handler = CreateHandler<THandler>(view);
+				return func(handler);
```

Comment:
What's happening here? Why is it no longer async and what does the func() do that's different?

Reply:
This is just a copy paste of the same method that's inside `Core/tests/DeviceTests/HandlerTestBase`

It's still all async, the Task is just being returned to the caller and the caller has to await the Task. The `func` is just the operation you want invoked on the main thread.